src/main/cpp/blaze.cc

```diff
@@ -1623,6 +1616,12 @@ int Main(int argc, const char *const *argv, WorkspaceLayout *workspace_layout,
   ParseOptionsOrDie(cwd, workspace, *option_processor, argc, argv);
   StartupOptions *startup_options = option_processor->GetParsedStartupOptions();
   startup_options->MaybeLogStartupOptionWarnings();
+  const string self_path = GetSelfPath(argv[0], *startup_options);
+
+  if (argc == 2 && strcmp(argv[1], "--version") == 0) {
```

Comment:
Unfortunately moving this check here breaks `bazel --version`, and also unfortunately the test for this only present internally. The complication is `--version` is a magic flag that isn't understood by the options parser. This is by design so that we traverse as little code (which could fail) as possible, including rc parsing.

I'm not sure what the best way to work around this would be... I think we'd either need to rethink how `--version` is implemented wrt the options parser or see if there's another way to toggle or fall back to the feature you're proposing.

Reply:
> I'm not sure what the best way to work around this would be...

If we're running into issues like this I'd recommend abandoning trying to use a startup flag. Instead, either:
* Use an environment variable, as proposed by @emidln-imc 
in https://github.com/bazelbuild/bazel/pull/14391#issuecomment-1022610305
* Have the client dynamically determine what to do, as I proposed in https://github.com/bazelbuild/bazel/pull/14391#issuecomment-1022710415

Reply:
+1 to dynamically detecting that if possible. One problem here potentially is that from what I was told, `realpath("/proc/self/exe")` points to Bazel, in which case I don't know how to distinguish that from `realpath(getauxval(AT_EXECFN))`. Does `readlink` offer a way to distinguish those 2?